### ISSUE TITLE
Cache autosave document to reduce writes to disk

### ISSUE THREAD

**Issue by juliusknorr** [nextcloud/text#1042]
- Temporarily dump the document to save into a cache
- Write it back less often than every 10s to reduce possible disk writes (also db load when triggering activity)

Downside: File changes might take longer to be represented in the actual files, which i'd say is acceptable since in most cases the editing will happen mainly in the text app then. We can still force save to disk when users actively close the session or press ctrl+s

Benefit: Less disk access

cc @rullzer

**Comment by jospoortvliet** [comment]
Less often than every 10 seconds would be good. Indefinitely is not - maybe a once-every-five-minutes or so, IF there are edits of course. You do want the changes to get synced to the clients, just not too often...

**Comment by juliushaertl** [comment]
Yeah we can basically persist it with the cronjob then, if no user is connected anymore, if users are active we then just save less regularly.

**Comment by mejo-** [comment]
Nowadays we only write the document every 30 seconds. I'll close this issue given that the topic didn't come up in the last years.
